torch/csrc/jit/tracer.h

```diff
@@ -64,34 +64,62 @@ inline std::vector<VariableFlags> getVarFlags(const variable_list& vars) {
 // are treated as constants.
 //
 // TODO: This code lives in the hotpath; make sure it is fast
-inline bool isTracing(const Variable& var) {
+inline bool isTracingVar(const Variable& var) {
   if (!var.defined() || !var.tracing_state()) return false;
   return std::any_of(var.tracing_state()->begin(), var.tracing_state()->end(), detail::isElemActive);
 }
 
-inline bool isTracing(std::initializer_list<Variable> vars) {
+inline bool isTracingVar(at::ArrayRef<Variable> vars) {
   // Reference to avoid refcount bump
-  for (auto& var : vars) {
-    if (isTracing(var)) return true;
+  for (const Variable& var : vars) {
+    if (isTracingVar(var)) return true;
   }
   return false;
 }
 
-inline bool isTracing(const at::ArrayRef<Variable>& vars) {
-  // Reference to avoid refcount bump
-  for (const Variable& var : vars) {
-    if (isTracing(var)) return true;
-  }
-  return false;
+inline bool isTracing(const at::Tensor& x) {
+  return isTracingVar(x);
+}
+
+inline bool isTracing(const at::Tensor& x, const at::Tensor& y) {
+  return isTracingVar({x, y});
+}
+
+inline bool isTracing(const at::Tensor& x, const at::Tensor& y, const at::Tensor& z) {
+  return isTracingVar({x, y, z});
+}
+
+inline bool isTracing(const at::Tensor& x, const at::Tensor& y, const at::Tensor& z, const at::Tensor& a) {
+  return isTracingVar({x, y, z, a});
+}
+
+inline bool isTracing(const at::Tensor& x, const at::Tensor& y, const at::Tensor& z, const at::Tensor& a, const at::Tensor& b) {
+  return isTracingVar({x, y, z, a, b});
 }
 
-inline bool isTracing(const at::TensorList& vars) {
+inline bool isTracing(const at::Tensor& x, const at::Tensor& y, const at::Tensor& z, const at::Tensor& a, const at::Tensor& b, const at::Tensor& c) {
+  return isTracingVar({x, y, z, a, b, c});
+}
+
+inline bool isTracing(const at::Tensor& x, const at::Tensor& y, const at::Tensor& z, const at::Tensor& a, const at::Tensor& b, const at::Tensor& c, const at::Tensor& d) {
+  return isTracingVar({x, y, z, a, b, c, d});
+}
+
+inline bool isTracing(at::TensorList vars) {
   for (const auto & var_t : vars) {
     if (isTracing(static_cast<const Variable&>(var_t))) return true;
   }
   return false;
 }
 
+inline bool isTracing(const at::Tensor& x, ArrayRef<at::Tensor> y) {
+  return isTracingVar(x) || isTracing(y);
+}
+
+inline bool isTracing(const at::Tensor& x, ArrayRef<at::Tensor> y, const at::Tensor& z) {
```

Comment:
omg what did even happen here. how did we go from 2 overloads to 12?! I don't think that's the right fix for any problem we had with the previous API. I'm still unsure what exactly was broken, let's chat about it.

Reply:
I'm replacing it with a template vararg.